### PR TITLE
WT-9047 Add tiered storage support to wtperf

### PR DESCRIPTION
**Pull request by sueloverso**
…er for a tiered storage workload.

### CODE REVIEW

**Comment by sueloverso** [comment]
Here's the diff to add tiered storage usage to `wtperf`. @ddanderson this is still one of the easiest pieces of software to extend and you originally wrote it. Thank you again. Adding a major piece of functionality that works in just a couple hours is a testament to this code's structure.

I believe this is ready. The commented out configurations drop core due to WT-7833. It mostly runs single-threaded.

File: bench/wtperf/wtperf.c [hunk 1]
```diff
@@ -1362,6 +1362,62 @@ checkpoint_worker(void *arg)
...
+            sleep(1);
+            if (wtperf->stop)
+                break;
+        }
+        /* If the workers are done, don't bother with a final call. */
+        if (wtperf->stop)
+            break;
+
+        wtperf->flush = true;
+        if ((ret = session->flush_tier(session, NULL)) != 0) {
```
Comment (keitharnoldsmith) [inline bench/wtperf/wtperf.c:1399]: Maybe include `sync=on`?  By analogy, `checkpoint_worker()` waits for the checkpoint to finish then waits for the `checkpoint_interval` to start the next checkpoint.  So maybe it sesms more natural for `flush_tier` to work the same way.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM.

One minor comment/suggestion.

File: bench/wtperf/wtperf.c (hunk 1 above)
Comment (sueloverso) [reply bench/wtperf/wtperf.c:1399]: `sync=on` is the default, do you think it needs to be explicit? Also, this new function was originally a copy of the `checkpoint_worker` function, and it is waiting the flush interval to start the next flush_tier. So it does work the exact same way as checkpoint. I must be misunderstanding your comment in some way. Can you be specific on what change you want to see regarding the interval?

**Review by ddanderson — APPROVED** [review]
LGTM.  One stray pathname to be fixed?

File: bench/wtperf/runners/tiered-read-only.wtperf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+conn_config="cache_size=20G,log=(enabled=true),tiered_storage=(bucket=/Users/sasha/Work/WiredTiger/BUCKET,bucket_prefix=pfx,local_retention=2,name=dir_store)"
```
[the remark this answers is not in the document]
Comment (sueloverso) [reply bench/wtperf/runners/tiered-read-only.wtperf]: Fixed! Thanks.  @fedorova I made it a relative path to `../BUCKET` and added in a few lines of commented out configuration items that I think you need in order to make tiered storage do something (i.e. have a flush interval and checkpoints).

File: bench/wtperf/wtperf.h [hunk 3]
```diff
@@ -67,6 +68,17 @@ typedef struct __truncate_queue_entry TRUNCATE_QUEUE_ENTRY;
 #define ZSTD_BLK BLKCMP_PFX "zstd"
 #define ZSTD_EXT EXT_PFX EXTPATH ZSTD_PATH EXT_SFX
 
+/* Tiered Storage Extensions */
+#ifndef DIR_STORE_PATH
+#define DIR_STORE_PATH "storage_sources/dir_store/libwiredtiger_dir_store.so"
+#endif
+#ifndef S3_PATH
+#define S3_PATH "storage_sources/s3_store/.libs/libwiredtiger_s3_store.so"
+#endif
```
Comment (fedorova) [inline bench/wtperf/wtperf.h:77]: @sueloverso , IIRC, the CMake build system does not put shared libraries into the .libs directory. So the S3_PATH should be: storage_sources/s3_store/libwiredtiger_s3_store.so

**Comment by sueloverso** [comment]
@ddanderson I requested a re-review from (only) you. Earlier commits were not passing PR testing. There is one test `conf-dump-test` that runs the `WT_TEST/CONFIG.wtperf` and that was badly constructed in that test (but not reproducible on any machine I tried). The nature of the bad path was this generated config for the connection:
`conn_config="cache_size=16GB,eviction=(threads_max=4),log=(enabled=false),session_max=33,,extensions=(../../ext/snappy/libwiredtiger_snappy.so),session_max=97"` .

This change modified the pathnames in `wtperf.h`. I have removed the `ifndef` from `wtperf.h` so that we generate a correct SNAPPY_PATH and overwrite it if it is defined.

**Pull request merged**